### ISSUE TITLE
Don't throw exception in svm.cpp

### ISSUE THREAD

**Issue by rosetta-jpn** [Netflix/vmaf#1051]
[svm.cpp](https://github.com/Netflix/vmaf/blob/master/libvmaf/src/svm.cpp) is only the place of throwing exception in libvmaf.
Looks like the code can be simply replaced with bool + error message.
Does it make sens to remove code so that libvmaf can be built with -fno-exceptions compiler option?

If this is agreed, I will send a pull-request.

Thanks in advance.

**Comment by kylophone** [comment]
> [svm.cpp](https://github.com/Netflix/vmaf/blob/master/libvmaf/src/svm.cpp) is only the place of throwing exception in libvmaf.
> Looks like the code can be simply replaced with bool + error message.
> Does it make sens to remove code so that libvmaf can be built with -fno-exceptions compiler option?

Yes, feel free to send a PR for this. `libsvm` is 3rd party code, but I think we are OK to patch this file.

>Actually, tools/vmaf.cpp also handles std::stoi error handling and also enclose the core vmaf computation function by "try".
I am not sure if vmaf calls some standard function throwing an exception except svm.cpp and std::stoi in main.cpp.

Don't worry about this one. This is an old tool, and should be removed from the codebase.

**Comment by rosetta-jpn** [comment]
Thanks for the info. I created a pull request for svm.cpp. https://github.com/Netflix/vmaf/pull/1071
Ideally, once the exceptions in main.cpp are removed, the compiler flag -fno-exceptions should be added to Makefile.

**Comment by rosetta-jpn** [comment]
Gentle ping.
